pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -293,6 +296,15 @@ func (c *Controller) validateTraceflow(tf *opsv1alpha1.Traceflow) error {
 	if tf.Spec.Destination.Service != "" && !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
 		return errors.New("using Service destination requires AntreaProxy feature enabled")
 	}
+	if tf.Spec.Destination.IP != "" {
+		destIP := net.ParseIP(tf.Spec.Destination.IP)
+		if destIP == nil {
+			return fmt.Errorf("destination IP is not valid: %s", tf.Spec.Destination.IP)
+		}
+		if c.serviceCIDR.Contains(destIP) && !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
```

Comment:
Just personal thinking so feel free to ignore - I feel it is easier to understand if we change to:
`!features.DefaultFeatureGate.Enabled(features.AntreaProxy) && c.serviceCIDR.Contains(destIP)`
because, when AntreaProxy is enabled, serviceCIDR is not a required parameter and the value might be incorrect.

Reply:
Makes sense to me. Updated the code and added a comment.